### ISSUE TITLE
How to test Webview with react-native-testing-library?

### ISSUE THREAD

**Issue by fabiendem** [callstack/react-native-testing-library#1255]
Hi,

I posted the question over at https://github.com/react-native-webview/react-native-webview/discussions/2689 but it didn't get much attention, so I am trying here just in case.

Copy paste:
> I am struggling to find good examples of tests while using the `react-native-webview` and `react-native-testing-library` under jest.
> How do you usually test components which render a webview? Especially, how do you simulate the successful loading of the webpage in the context of the test renderer?

Thanks!

**Comment by mdjastrzebski** [comment]
TLDR: You will need to create Jest mock for WebView component.

RNTL is not able to run native part of your RN app. Only the JS part. You can read more details [here](https://callstack.github.io/react-native-testing-library/docs/faq) and [here](
https://callstack.github.io/react-native-testing-library/docs/testing-env#react-test-renderer).

In some situations RN libraries with native code provide their own Jest mocks. It seems like RN WebView might not be one of these libraries. In such case you will need to write your own Jest mocks, which is not that hard. Here is a [relevant doc](https://jestjs.io/docs/tutorial-react-native#mock-native-modules-using-jestmock) from Jest. You will probably need to mock some lifecycle events of WebView and/or `postMessage` stuff with `jest.fn()`.

If you manage to make it work for your tests I invite you to share a generic mock with us, e.g. in these issue, so that we can improve our documentation with it.

**Comment by fabiendem** [comment]
Thank you @mdjastrzebski, that's very clear.
I will share this with the team.
Feel free to close the ticket.
